File: android/src/main/java/dragon3/data/WazaData.java
package dragon3.data;

import java.util.ArrayList;
import java.util.List;

import dragon3.common.constant.AttackEffect;
import dragon3.common.constant.DamageType;
import dragon3.common.constant.EnergyType;
import dragon3.common.constant.GameColor;
import dragon3.common.constant.TargetType;

@lombok.Data
public class WazaData implements Data {

	private String id = "none";
	private String name = "none";
	private String label = "none";
	private String image = "none.png";
	private GameColor labelColor = GameColor.NONE;

	private DamageType damageType = DamageType.NONE;
	private TargetType targetType = TargetType.SINGLE_1;
	private String animeId = "none";
	private EnergyType energyType = EnergyType.NONE;
	private int energyCost;

	private List<AttackEffect> effect = new ArrayList<>();

	@Override
	public String toString(){
		return id + " - " + name;
	}
}

File: core/src/main/kotlin/dragon3/data/WazaData.kt
package dragon3.data

import java.util.ArrayList

import dragon3.common.constant.AttackEffect
import dragon3.common.constant.DamageType
import dragon3.common.constant.EnergyType
import dragon3.common.constant.GameColor
import dragon3.common.constant.TargetType

class WazaData : Data {

    override var id = "none"
    override var name = "none"
    var label = "none"
    var image = "none.png"
    var labelColor = GameColor.NONE

    var damageType = DamageType.NONE
    var targetType = TargetType.SINGLE_1
    var animeId = "none"
    var energyType = EnergyType.NONE
    var energyCost: Int = 0

    var effect = ArrayList<AttackEffect>()

    override fun toString(): String {
        return id + " - " + name
    }
}
